core/src/test/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -1576,7 +1586,7 @@ private void removeDisruptionSchemeFromNode(NodeAndClient nodeAndClient) {
         return Collections2.filter(nodes.values(), new DataOrMasterNodePredicate());
     }
 
-    private static final class DataNodePredicate implements Predicate<NodeAndClient> {
+    public static final class DataNodePredicate implements Predicate<NodeAndClient> {
```

Comment:
see above, added a new method for restart so that I don't need the predicates anymore